### DISCUSSION TITLE
[Feature Request] Run button on the  selected line

### DISCUSSION THREAD

**Discussion by kartik-gupta-ij** [microsoft/monaco-editor#3798] [Q&A]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

 
![image](https://user-images.githubusercontent.com/84975264/224562260-cd81305b-b1dc-40e6-8488-7eb5565908fb.png)

something similar to this

**Comment by tamayika** [comment]
Please see [this example](https://microsoft.github.io/monaco-editor/playground.html?source=v0.36.1#example-interacting-with-the-editor-listening-to-mouse-events).
You can show custom DOM as overlay widget.

**Reply by kartik-gupta-ij** [reply]
1. How to use onClick event listener ?

2. How to move the overlay widget just at the end of the line  where the cursor is ?

**Reply by tamayika** [reply]
You are creating DOM by yourself, so you can set event listener.
```ts
var domNode = document.createElement("div");
domNode.addEventListener("click", () => console.log("clicked"));
```

**Reply by kartik-gupta-ij** [reply]
> 2\. How to move the overlay widget just at the end of the line  where the cursor is ?

what about this ?

**Reply by tamayika** [reply]
Please see below customized example link.

https://microsoft.github.io/monaco-editor/playground.html?source=v0.36.1#XQAAAAJGBQAAAAAAAABBqQkHQ5NjdZ76Sv_fvV7aSEHtH3wvJgbIGfhhbNxsZorBCr68QX8cj7riltqfDh-psYMjmtWwnRYsGiPX26BJejXF_mIjvZQ2fKjeLNnP8nt-t45EiLCJZMSAfWWvU5C8yZgTjkPOUTss_WG0kKvQqCw5KJnRLUYXn9yCfEpJ3m5ddv-XaS-6zrhZnA-EHwIJIxDuVyyDj57-NnkftIoUQiSYTkG4X2RZ-ctTDlz7KLYgMcMy6dUVrmArVpfFBEl57JPqXdjOUaa24f0iTAsKlvcoA5vGRdR7ElhwB7gq3acFjieDaXJ0TuCKNUHgZ9gRBhat5LF0xOALjBXgfZTvqx1xqynbKiFlFwAOmvraozMznEGVqKluScFdJHI4mEBAfVhddrjEILWHs-Eb_8Ew9mJtMI6gVitaI0oFndHfKLcimKwwCsr8a5zygBwPPXJsiMsS9qNnH0Y780akRuf_qZRzsMGOuPSu0yG3HVRZphHGuF0BGKAkpvcwIou5084rt6apQV0b7HGKlbvvcn2SFYLkWXAgQ601-Fc_Vbt1gtwleX6A91GIZYdUIIWAVmo__MWbpRIpA58TbwKP5atuCJk7i_8xhybRcNRjszsnBaNA4evCGy8GXp8IOqyK1q7JVaC_TU_k4O3G2gBQ7ZQghOXYvDbggfOfGLLdDv4GSuyqnAyDvcEkR8HBOxtvDOS0PSWiWWlNg0UL65P4B4W9OxelAEPjZytxi-rYdX5lfBdcv_5uSXbjdsgyISPpUrsKRELIxC3ByNtE9aV8mCmeMrjMZ8cxEkGnaWcaNN_8h1WCUDXm08Tho7yL7i-eJCGgQWESNf--fmu8Xny8GPPZ8W1qtxZAsfyk0KloGJD99ff9grgSurbTM6C5g35LQoNx_ITon-AJMcWqhIfJP4A9PvdeDBqngokBKO5WmGAYj90JwbAIWIcP9cmHkLESGHWJH9H_9aflqw

**Reply by kartik-gupta-ij** [reply]
Thanks, a lot 🙌
It helped me
